src/transformers/models/bit/modeling_bit.py

```diff
@@ -405,33 +462,42 @@ class BitBottleneckLayer(nn.Module):
 
     def __init__(
         self,
+        config,
         in_channels,
         out_channels=None,
         bottle_ratio=0.25,
         stride=1,
         dilation=1,
         first_dilation=None,
         groups=1,
-        act_layer=None,
         conv_layer=None,
-        norm_layer=None,
-        proj_layer=None,
         drop_path_rate=0.0,
+        is_first_layer=False,
     ):
         super().__init__()
         first_dilation = first_dilation or dilation
-        act_layer = act_layer or nn.ReLU
-        conv_layer = conv_layer or StdConv2d
-        norm_layer = norm_layer or partial(BitGroupNormActivation, num_groups=32)
+
+        # Getting the convolution type
+        if config.conv_layer == "std_conv":
+            conv_layer = partial(StdConv2d, eps=1e-8)
+        elif config.conv_layer == "std_conv_same":
+            conv_layer = partial(StdConv2dSame, eps=1e-8)
+        else:
+            raise ValueError(
+                f"Convolutional layer {config.conv_layer} not supported! Please use one of the following:"
+                " [`'std_conv'`, `'std_conv_same`]"
+            )
```

Comment:
We can probably improve this as I see this if else statement is defined 3 times in the modeling file

Reply:
Should be fixed in [37bfdfb](https://github.com/NielsRogge/transformers/pull/48/commits/37bfdfb9b24e8532414837ea2ac210753a6e7b87) - also refactored the Bottleneck layer